File: Leetcode/Python/977. Squares of a Sorted Array/solution.py
# https://leetcode.com/problems/squares-of-a-sorted-array/

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:  # Runtime: 254 ms
        nums = [num ** 2 for num in nums]
        return sorted(nums)
    def sortedSquares(self, nums: List[int]) -> List[int]:  # Runtime: 208 ms
        for i in range(len(nums)):
            nums[i] *= nums[i]
        nums.sort()
        return nums
            

File: Leetcode/Javascript/977. Squares of a Sorted Array/solution.js
// https://leetcode.com/problems/squares-of-a-sorted-array/

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
	for (let idx = 0; idx < nums.length; idx++) {
		nums[idx] **= 2;
	}
	nums.sort((a, b) => a - b);
	return nums;
};
